I2C Driver does not compile for nrf5 boards

When enabling I2C driver for nrf52_pca10040 or nrf51_pca10028 the following error is given:

```
zephyr/include/generated/generated_dts_board.h:91:33: error: 'NORDIC_NRF5_I2C0_40003000_BASE_ADDRESS_0' undeclared here (not in a function); did you mean 'NORDIC_NRF_UART_40002000_BASE_ADDRESS_0'?
zephyr/include/generated/generated_dts_board.h:93:31: error: 'NORDIC_NRF5_I2C0_40003000_CLOCK_FREQUENCY' undeclared here (not in a function); did you mean 'NORDIC_NRF5_I2C0_40003000_BASE_ADDRESS_0'?
zephyr/include/generated/generated_dts_board.h:92:28: error: 'NORDIC_NRF5_I2C0_40003000_LABEL' undeclared here (not in a function); did you mean 'NORDIC_NRF_UART_40002000_LABEL'?
```
And so on.

Tested on version c0d0a61bcc4f
To reproduce, build `samples/bluetooth/beacon/` for either nrf52_pca10040 or nrf51_pca10028 with the following configuration set:

CONFIG_I2C=y
CONFIG_I2C_NRF5=y
CONFIG_I2C_0=y
CONFIG_I2C_NRF5_0_GPIO_SDA_PIN=12
CONFIG_I2C_NRF5_0_GPIO_SCL_PIN=13

--- a/boards/arm/nrf51_pca10028/nrf51_pca10028.dts
+++ b/boards/arm/nrf51_pca10028/nrf51_pca10028.dts
@@ -24,6 +24,14 @@
 	status = "ok";
 };
 
+&i2c0 {
+	status = "ok";
+};
+
+&i2c1 {
+	status = "ok";
+};
+
 &flash0 {
 	/*
 	 * For more information, see:

--- a/boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts
+++ b/boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts
@@ -25,6 +25,14 @@
 	status = "ok";
 };
 
+&i2c0 {
+	status = "ok";
+};
+
+&i2c1 {
+	status = "ok";
+};
+
 &flash0 {
 	/*
 	 * For more information, see:

--- a/boards/arm/nrf52_pca10040/nrf52_pca10040.dts
+++ b/boards/arm/nrf52_pca10040/nrf52_pca10040.dts
@@ -25,6 +25,13 @@
 	compatible = "nordic,nrf-uart";
 	current-speed = <115200>;
 };
+&i2c0 {
+	status = "ok";
+};
+
+&i2c1 {
+	status = "ok";
+};
 
 &flash0 {
 	/*

